src/sentry/api/utils.py

```diff
@@ -377,10 +381,12 @@ def handle_query_errors() -> Generator[None]:
     except SnubaError as error:
         message = "Internal error. Please try again."
         arg = error.args[0] if len(error.args) > 0 else None
+        if isinstance(error, RateLimitExceeded):
+            sentry_sdk.set_tag("query.error_reason", "RateLimitExceeded")
```

Comment:
could also be worth capturing the exception here like we do for https://github.com/getsentry/sentry/blob/3e40d1c55ce747dc4cc81f10cb703b3c00636803/src/sentry/api/handlers.py#L13, as i find with sampling most of the time debugging a particular customer case it's not there in tracing

Reply:
I like that suggestion!